sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -1431,7 +1433,8 @@ def _restore_table_ref(sharded_table_ref_elems_kv):
             bigquery_tools.AppendDestinationsFn(self.table_reference),
             *self.table_side_inputs)
         | 'AddInsertIds' >> beam.ParDo(_StreamToBigQuery.InsertIdPrefixFn())
-        | 'ToHashableTableRef' >> beam.Map(_to_hashable_table_ref))
+        | 'ToHashableTableRef' >> beam.Map(_to_hashable_table_ref)
+    ).with_output_types(Tuple[str, Any])
```

Comment:
The explicit `Any` discards type information. Could you try annotating the types in `_to_hashable_table_ref` instead?

```py
V = TypeVar('V')

def _to_hashable_table_ref(table_ref_elem_kv: Tuple[Union[str, TABLE_REF_TYPE], V]) -> Tuple[str, V]:
```

Reply:
Thanks for the suggestion! Made the change accordingly.